ThirdPartyAdapters/unity/unity/src/main/java/com/google/ads/mediation/unity/UnityMediationAdapter.java

```diff
@@ -209,83 +127,29 @@ public void initialize(Context context,
+            }
+
+            @Override
+            public void onInitializationFailed(UnityAds.UnityAdsInitializationError
+                                                           unityAdsInitializationError, String s) {
+                initializationCompleteCallback.onInitializationFailed("Initialization failed: " + s);
+            }
+        });
     }
 
     @Override
     public void loadRewardedAd(MediationRewardedAdConfiguration mediationRewardedAdConfiguration,
                                MediationAdLoadCallback<MediationRewardedAd,
                                        MediationRewardedAdCallback> mediationAdLoadCallback) {
-        Context context = mediationRewardedAdConfiguration.getContext();
-        if (!(context instanceof Activity)) {
-            mediationAdLoadCallback.onFailure("Context is not an Activity." +
-                    " Unity Ads requires an Activity context to show ads.");
-            return;
-        }
-
-        Bundle serverParameters = mediationRewardedAdConfiguration.getServerParameters();
-        String gameID = serverParameters.getString(KEY_GAME_ID);
-        mPlacementId = serverParameters.getString(KEY_PLACEMENT_ID);
-
-        if (!isValidIds(gameID, mPlacementId)) {
-            mediationAdLoadCallback.onFailure("Failed to load ad from UnityAds: " +
-                    "Missing or invalid game ID and placement ID.");
-            return;
-        }
-
-        mMediationAdLoadCallback = mediationAdLoadCallback;
-
-        UnitySingleton.getInstance().initializeUnityAds((Activity) context, gameID);
-        UnitySingleton.getInstance().loadAd(mUnityAdapterRewardedAdDelegate);
-    }
-
-    @Override
-    public void showAd(Context context) {
-        if (!(context instanceof Activity)) {
-            String message = "An activity context is required to show Unity Ads.";
-            Log.w(TAG, message);
-            if (mMediationRewardedAdCallback != null) {
-                mMediationRewardedAdCallback.onAdFailedToShow(message);
-            }
-            return;
-        }
-        Activity activity = (Activity) context;
-
-        // Request UnitySingleton to show video ads.
-        if (UnityAds.isReady(mPlacementId)) {
-            UnitySingleton.getInstance().showAd(mUnityAdapterRewardedAdDelegate, activity);
-
-            // Unity Ads does not have an ad opened callback.
-            if (mMediationRewardedAdCallback != null) {
-                mMediationRewardedAdCallback.onAdOpened();
-                mMediationRewardedAdCallback.reportAdImpression();
-            }
-        } else {
-            if (mMediationRewardedAdCallback != null) {
-                mMediationRewardedAdCallback.onAdFailedToShow(
-                        "UnityAds placement '" + mPlacementId + "' is not ready.");
-            }
-        }
-
+        rewardedAd = new UnityRewardedAd(mediationRewardedAdConfiguration,
+                mediationAdLoadCallback);
+        rewardedAd.load();
```

Comment:
we simply can use a separate instance of a listener instead. The same way is Facebook does.

I know that I told to make things like that, but I didn't realize a possible impact.

keeping similar structure as `private UnityAdapterDelegate mUnityAdapterRewardedAdDelegate ...` (above) would be better...